File: content/snippets/typescript/design_patterns/lazy_initialization.ts
class LazySingleton {
  private static _instance: LazySingleton | null = null
  private constructor(public value = 0){}
  static instance(){
    if (!this._instance) this._instance = new LazySingleton()
    return this._instance
  }
}
console.log(LazySingleton.instance().value)
export {}

File: content/snippets/python/design_patterns/lazy_initialization.py
class LazySingleton:
    _instance = None
    def __init__(self):
        self.value = 0
    @classmethod
    def instance(cls):
        if cls._instance is None:
            cls._instance = LazySingleton()
        return cls._instance

if __name__ == '__main__':
    LazySingleton.instance().value = 9
    print(LazySingleton.instance().value)
